--- a/Makerules
+++ b/Makerules
@@ -784,0 +785,5 @@
+if test "$(@F)" = libc_pic.a; then \
+  cd $(common-objdir) && for stamp in $(patsubst $(common-objpfx)%,%,$^); do \
+    cp "$$stamp" "$$stamp.saved"; \
+  done; \
+fi
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -247,4 +247,4 @@
-	    LC_ALL=C fgrep -l /$$file \
-		  $(common-objpfx)stamp.os $(common-objpfx)*/stamp.os | \
-	    LC_ALL=C \
-	    sed 's@^$(common-objpfx)\([^/]*\)/stamp\.os$$@rtld-\1'" +=$$file@"\
+	    LC_ALL=C fgrep -l /$$file \
+		  $(common-objpfx)stamp.os.saved $(common-objpfx)*/stamp.os.saved | \
+	    LC_ALL=C \
+	    sed 's@^$(common-objpfx)\([^/]*\)/stamp\.os\.saved$$@rtld-\1'" +=$$file@"\
