--- a/elf/Makefile
+++ b/elf/Makefile
@@ -247,4 +247,10 @@
-	    LC_ALL=C fgrep -l /$$file \
-		  $(common-objpfx)stamp.os $(common-objpfx)*/stamp.os | \
-	    LC_ALL=C \
-	    sed 's@^$(common-objpfx)\([^/]*\)/stamp\.os$$@rtld-\1'" +=$$file@"\
+	    static_file=`echo "$$file" | sed 's/\.os$$/.o/'`; \
+	    for stamp in `find $(common-objpfx) -name stamp.o -type f`; do \
+	      LC_ALL=C fgrep -q /$$static_file "$$stamp" && echo "$$stamp"; \
+	    done | LC_ALL=C \
+	    sed 's@^$(common-objpfx)\([^/]*\)/stamp\.o$$@rtld-\1'" +=$$file@"; \
+	    nonshared_file=`echo "$$file" | sed 's/\.os$$/.oS/'`; \
+	    for stamp in `find $(common-objpfx) -name stamp.oS -type f`; do \
+	      LC_ALL=C fgrep -q /$$nonshared_file "$$stamp" && echo "$$stamp"; \
+	    done | LC_ALL=C \
+	    sed 's@^$(common-objpfx)\([^/]*\)/stamp\.oS$$@rtld-\1'" +=$$file@" \
